### ISSUE TITLE
[Feature Request] Support for Xbox Wireless headset

### ISSUE THREAD

**Issue by akuropka** [NiyaShy/XB1ControllerBatteryIndicator#73]
Hi @NiyaShy,
first of all, it's great of you developing this nice tool further. It works like a charm with my controller.
I am wondering wether it would be possible to have the battery level of a headset included as well. I have both connected through the wireless adapter.
Cheers,
Andreas

**Comment by NiyaShy** [comment]
Hey,
there are headsets you can pair with the proprietary MS dongle? Haven't heard of that yet...
The thing is: even if they work with the dongle, since they are meant to transmit audio it's very likely that they operate on some protocol that has nothing to do with XInput, if that's the case it would mean implementing a whole different way of polling the battery level. And since you're the first person asking about it, that type of headset doesn't seem to be very widespread. So for now, it's a "No" from me.

If there's proper documentation about how that type of headset operates AND more people request it, that would increase the chance of a change of heart 😉

**Comment by akuropka** [comment]
The MS Xbox wireless headset can be paired with the dongle, by Bluetooth or wired USB connection, same as the controller. The Xbox app as well as the devices app do show the battery level. That’s why I was thinking the routine is somehow similar but it was just a guess.

**Comment by cascaabreu** [comment]
Hello there, also an user of the Xbox Wireless Headset with the Xbox Wireless dongle.
You are not alone @akuropka 👍

**Comment by NiyaShy** [comment]
Hey,
just looked into it a bit more, and I couldn't find any technical info about how the headset communicates (aside from that it uses the same proprietary transmission protocol the controllers use), so I guess Microsoft hasn't published that (yet). and without info, I can't look into if it would be possible to query the battery level.

A good (or in that context, bad) sign that MS is keeping the details secret is that another open source project, [xone](https://github.com/medusalix/xone), a driver for getting the proprietary dongles to work under linux, has so far only figured out how to get controllers working via the dongle, not headsets. Can't do much unless MS has the mercy of releasing some tech details so others can figure out the protocols.

tl;dr: Microsoft keeps how the headset-dongle combo works a secret, can't do anything about it.
